lib/common/grpc-service.js

```diff
@@ -266,6 +268,25 @@ GrpcService.prototype.request = function(protoOpts, reqOpts, callback) {
     request: function(_, onResponse) {
       respError = null;
 
+      if (stream) {
+        service[protoOpts.method](reqOpts, grpcOpts)
+          .pipe(stream)
```

Comment:
Can I see the code you're using that triggers the errors? In my own testing, stream-cache seems to support anything:

```js
var cacheStream  = new StreamCache();
var readStream = through.obj();

readStream.pipe(cacheStream);

cacheStream.pipe(through.obj(function(obj, e, next) {
  console.log(obj)
  next()
}));

readStream.push({ a: 'b' });
readStream.end();
```

(logs: `{ a: 'b' }`)